x/fees/keeper/evm_hooks.go

```diff
@@ -0,0 +1,80 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+)
+
+// Hooks wrapper struct for fees keeper
+type Hooks struct {
+	k Keeper
+}
+
+var _ evmtypes.EvmHooks = Hooks{}
+
+// Hooks return the wrapper hooks struct for the Keeper
+func (k Keeper) Hooks() Hooks {
+	return Hooks{k}
+}
+
+// PostTxProcessing implements EvmHooks.PostTxProcessing. After each successful
+// interaction with a registered contract, the contract deployer receives
+// a share from the transaction fees paid by the user.
+func (h Hooks) PostTxProcessing(ctx sdk.Context, msg core.Message, receipt *ethtypes.Receipt) error {
+	// check if the fees are globally enabled
+	params := h.k.GetParams(ctx)
+	if !params.EnableFees {
+		return nil
+	}
+
+	contract := msg.To()
+	// if the contract is not registered to receive fees, do nothing
+	if contract == nil || !h.k.IsFeeRegistered(ctx, *contract) {
+		return nil
+	}
+
+	cfg, err := h.k.evmKeeper.EVMConfig(ctx)
```

Comment:
do you need this for something other than the EVM denom? I guess we can query the EVM params instead of getting the full `EVMConfig`. What do you think?